### PR TITLE
(BKR-1113) do_scp_to now also copies files that start with a dot

### PR DESCRIPTION
**Pull request by pegasd**
With this patch do_scp_to also copies all files that start with a dot (this issue was noticed when using copy_module_to).

The fixed pattern is taken from glob() documentation (https://ruby-doc.org/core-2.4.1/Dir.html#method-c-glob).

### CODE REVIEW

File: lib/beaker/host.rb [hunk 1]
```diff
@@ -438,7 +438,7 @@ def do_scp_to source, target_path, options
           @logger.trace result.stdout
         end
       else # a directory with ignores
-        dir_source = Dir.glob("#{source}/**/*").reject do |f|
+        dir_source = Dir.glob("#{source}/**/{*,.*}").reject do |f|
```
Comment (ferglor) [inline lib/beaker/host.rb]: Can you update this PR with tests for this?

Comment (epivenpro) [inline lib/beaker/host.rb]: I gave it a go in the next commit.
However, the spec tests pass with both versions of the code, which is probably due to FakeFS implementation in respect to `glob()`'s behavior.
Unfortunately, getting the acceptance suite going was a little outside of scope of my weekend endeavour, but the test should look similar to this.

Comment (pegasd) [inline lib/beaker/host.rb]: I gave it a go in the next commit.
However, the spec tests pass with both versions of the code, which is probably due to FakeFS implementation in respect to `glob()`'s behavior.
Unfortunately, getting the acceptance suite going was a little outside of scope of my weekend endeavour, but the test should look similar to this.

**Comment by tvpartytonight** [comment]
@puppetlabs-jenkins test this please

File: lib/beaker/host.rb (hunk 1 above)
Comment (kevpl) [inline lib/beaker/host.rb]: @pegasd this looks like a good change, but we'd like to not change the default behavior, but instead have this guarded behind an option like (for example) `:copy_hidden_files` for the method.

**Comment by tvpartytonight** [comment]
@pegasd last comment still relevant.

**Comment by kevpl** [comment]
@pegasd last comment is still current. This PR will be closed in a week if there's no activity.

File: lib/beaker/host.rb (hunk 1 above)
Comment (pegasd) [inline lib/beaker/host.rb]: Was away for a while, just now catching up.
@kevpl Is there a simple example I can dig into to hide this behavior behind an option?

Comment (pegasd) [inline lib/beaker/host.rb]: Also, how will you use that option from a puppet module when running tests?

**Pull request closed without merging**